### PR TITLE
feature/create-redux-store

### LINKED ISSUE #1: OHC-1 Create initial reducers and state slices

### PR DESCRIPTION
**Pull request by chooidos**
- added redux
- setup redux using redux-toolkit

### CODE REVIEW

File: src/modules/oh-indicators-list/List.test.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { List } from './List';
+
+describe('ItemsModule', () => {
+  it('should fetch list of all the items when rendered', () => {
+    render(<List />);
+    expect(screen.queryAllByRole('listitem').length).not.toBe(0);
```
Comment (dmylian) [inline src/modules/oh-indicators-list/List.test.tsx]: @chooidos change it to the real value:
`expect(...).toBe(xx);`

File: src/modules/oh-indicators-list/store/slice.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import { IOpenhabItem } from '../types/openHab';
+
+export interface IState {
+  items: IOpenhabItem[];
+  error: string | null;
+}
+
+export const getAllItems = createAsyncThunk<IOpenhabItem[], void>(
+  'items/fetchAllItems',
+  fetchAllItems,
```
Comment (dmylian) [inline src/modules/oh-indicators-list/store/slice.ts]: as we don't do anything with the value,
```
async () => {
  await getData();
}
```
can be simplified into a 
```
getData
```

File: src/modules/oh-indicators-list/store/slice.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+);
+
+export const indicatorsSlice = createSlice<IState, any>({
+  name: 'persistentStateSlice',
+  initialState: {
+    items: [],
+    error: null,
```
Comment (dmylian) [inline src/modules/oh-indicators-list/store/slice.ts:19]: added error field into the state, cause all I see in the app are network errors

File: src/store.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { configureStore } from '@reduxjs/toolkit';
+import persistantStateSlice from './modules/oh-indicators-list/store/slice';
+
+export const store = configureStore({
+  reducer: {
+    indicators: persistantStateSlice,
```
Comment (dmylian) [inline src/store.ts:6]: Renamed to `indicators` cause `persistent` is too generic. 
Open to a discussion

File: package.json [hunk 5]
```diff
@@ -3,15 +3,20 @@
...
   "dependencies": {
+    "@reduxjs/toolkit": "^1.6.0",
     "@testing-library/jest-dom": "^5.14.1",
     "@testing-library/react": "^11.2.7",
     "@testing-library/user-event": "^12.8.3",
     "@types/jest": "^26.0.23",
     "@types/node": "^12.20.15",
     "@types/react": "^17.0.11",
     "@types/react-dom": "^17.0.8",
+    "@types/react-router-dom": "^5.1.8",
```
Comment (dmylian) [inline package.json:14]: Only wrapped single exisintg module into a `/`

**Comment by dmylian** [comment]
@chooidos 

- [x] review the proposed folder structure and the changes I've made
- [x] update the tests to reflect the real data
- [x] think about how can the local network be shared. ~~Maybe [expose.dev](https://expose.dev/)?~~
- [x] create a local env file to override the default https://github.com/chooidos/OH-Card/pull/2#discussion_r669046988
- [x] verify that data loads and being saved to the store

File: src/modules/oh-indicators-list/store/slice.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  },
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(getAllItems.fulfilled, (state, action: PayloadAction<IOpenhabItem[]>) => {
+      state.items = state.items.concat(action.payload);
```
Comment (dmylian) [inline src/modules/oh-indicators-list/store/slice.ts]: @chooidos think about which data structure is easier to access by an element id? Is array (current) good for that?

File: .env [hunk 7]
```diff
@@ -0,0 +1 @@
+REACT_APP_BASE=https://176.241.137.67:18443
```
Comment (dmylian) [inline .env]: @chooidos 
https://create-react-app.dev/docs/adding-custom-environment-variables/#adding-development-environment-variables-in-env
https://tacomanator.medium.com/environments-with-create-react-app-7b645312c09d
create an `.env.local` to override with `http://192.168.88.150:8080`, as it's your local network

Comment (chooidos) [reply .env]: I think it should be changeable from application

Comment (dmylian) [reply .env]: why? is it supposed to connect to different backends during its lifetime? Pretty sure we can statically assign different envs that won't need to be changed

Comment (chooidos) [reply .env]: it will be static in my local network, but ISP could change external IP address.

Comment (chooidos) [reply .env]: also, is it safe to store IP addresses in repo?

Comment (dmylian) [reply .env]: got it
yes, good point, it's not very safe. Can you safely delete it?
https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository

**Review by dmylian — APPROVED** [review]

**Pull request merged**